### PR TITLE
Added experimental support for legacy Java 1.6 JVMs through Retrolambda.

### PR DESCRIPTION
**Pull request by ubschmidt2**

### CODE REVIEW

File: contrib/agent/build.gradle [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 }
 
 description = 'OpenCensus Agent'
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = JavaVersion.VERSION_1_7
+targetCompatibility = JavaVersion.VERSION_1_7
+
+// Experimental support for legacy Java 1.6 JVMs.
```
Comment (adriancole) [inline contrib/agent/build.gradle]: might want to consider adding animal sniffer to ensure later apis are not used

Comment (ubschmidt2) [inline contrib/agent/build.gradle]: IIUIC, the root project takes care of this (animalsniffer) already. Signature for the agent is 
signature 'org.codehaus.mojo.signature:java16:+@signature'

**Comment by ubschmidt2** [comment]
Now fails to build with JDK 7 because Retrolambda wants JDK 8. Investigating...

**Comment by adriancole** [comment]
ah maybe worth a comment saying this.

**Comment by ubschmidt2** [comment]
So Retrolambda itself needs a JDK 8 for doing its magic. It uses $JAVA8_HOME, which I've added to the Travis config.

**Pull request merged**